tests/cupy_tests/statics_tests/test_histogram.py

```diff
@@ -148,3 +148,138 @@ def test_bincount_zero(self, xp, dtype):
     def test_bincount_too_small_minlength(self, xp, dtype):
         x = testing.shaped_arange((3,), xp, dtype)
         return xp.bincount(x, minlength=-1)
+
+
+@testing.gpu
+@testing.parameterize(*testing.product(
+    {'bins': [
+        # Test monotonically increasing with in-bounds values
+        [0, 1, 2, 4, 10],
+        [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        [0.0, 1.0, 2.5, 4.0, 10.0],
+        [1.5, 2.5, 4.0, 6.0],
+        # Explicit out-of-bounds for x values
+        [-1.0, 1.0, 2.5, 4.0, 20.0],
+        # Repeated values should yield right-most or left-most indexes
+        [0.0, 1.0, 1.0, 4.0, 4.0, 10.0],
+        [0.0, 1.0, 1.0, 4.0, 4.0, 4.0, 4.0, 10.0],
+        # Monotonically decreasing arrays following above cases
+        [10.0, 4.0, 2.5, 1.0, 0.0],
+        [10.0, 4.0, 4.0, 2.5, 1.0, 0.0],
+        [4.0, 3.0, 2.5, 1.0],
+        [10.0, 10.0, 4.0, 4.0, 4.0, 4.0, 2.5, 1.0, 0.0, 0.0],
+        [15.0, 4.0, 2.5, 1.0, -1.0],
```

Comment:
Instead of explicitly specifying reverse-order bins, how about parameterize the order (increasing / decreasing)?